<!DOCTYPE HTML>
<!--
	Phantom by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>-10 and -35 Promoter Modification</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
		<noscript><link rel="stylesheet" href="assets/css/noscript.css" /></noscript>
	</head>
	<body class="is-preload">
		<!-- Wrapper -->
			<div id="wrapper">

				<!-- Header -->
					<header id="header">
						<div class="inner">

							<!-- Nav -->
								<nav>
									<ul>
										<li><a href="#menu">Menu</a></li>
									</ul>
								</nav>

						</div>
					</header>

				<!-- Menu -->
				<nav id="menu">
					<h2>Menu</h2>
					<ul>
						<li><a href="index.html">Projects</a></li>
						<li><a href="https://github.com/Rhys-sg">GitHub</a></li>
						<li><a href="https://medium.com/@sorensor">Medium</a></li>
						<li><a href="about.html">About</a></li>						
					</ul>
				</nav>

				<!-- Main -->
					<div id="main">
						<div class="inner">
							<h1>Promoter Modification</h1>
							<span class="image header">
								<img src="images/genetic_engineering.jpg" style="width: 1180px;" />
							</span>
							<table>
								<tr>
									<td>Description</td>
									<td>
										<p>This is a continuation of Predictive Artificial Neural Networks for mRNA Transcriptional Profiles and Sequence Motifs. For more information, see the <a href="sigma70predictiontool.html">description</a> and <a href="https://github.com/Rhys-sg/promoter-sequence-transcription-rates">GitHub repository</a>.</p>
										<p>This research is ongoing and in the extension to my previous project, I will be working with Professor Matthew Tien in the Whitman College Biology department.</p>
									</td>
								</tr>
								<tr>
									<td style="white-space: nowrap;">Research Stages</td>
									<td>
										<p>I developed a deep-learning model to predict E. coli transcription using multiple publicly available datasets, alongside a graphical user interface and relational database for students to upload gene expression data from synthetic promoters they designed. I collaborated in creating plasmids and a molecular biology workflow to help undergraduate students test their promoter designs. I also advised two professors on how to integrate deep-learning models into the curriculum, enabling students to explore the relationship between promoters and phenotypes through a biophysical model. I optimized convolutional neural networks to predict mRNA transcriptional initiation rates and σ70 promoter sequence motifs, using data from eleven publications with diverse methods for measuring transcriptional initiation. This optimization reduced the mean squared error (MSE) by 44% compared to previous machine learning models. I further developed a machine learning-based teaching tool for the Whitman BIOL-102 lab, creating its frontend, backend, and database architecture. For model validation and further training, I engineered E. coli σ70 promoter sequence motifs using RFP expression and developed a model to convert RFP expression into changes in Gibbs free energy.</p>
									</td>
								</tr>
							</table>
						</div>
					</div>

				<!-- Footer -->
				<footer id="footer">
					<div class="inner">							
						<ul class="copyright">
							<li>Design by <a href="http://html5up.net">HTML5 UP</a></li>
						</ul>
					</div>
				</footer>

			</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>