<!DOCTYPE HTML>
<!--
	Phantom by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>Sigma 70 Prediction Tool</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
		<noscript><link rel="stylesheet" href="assets/css/noscript.css" /></noscript>
	</head>
	<body class="is-preload">
		<!-- Wrapper -->
			<div id="wrapper">

				<!-- Header -->
					<header id="header">
						<div class="inner">

							<!-- Nav -->
								<nav>
									<ul>
										<li><a href="#menu">Menu</a></li>
									</ul>
								</nav>

						</div>
					</header>

				<!-- Menu -->
				<nav id="menu">
					<h2>Menu</h2>
					<ul>
						<li><a href="index.html">Projects</a></li>
						<li><a href="https://github.com/Rhys-sg">GitHub</a></li>
						<li><a href="https://medium.com/@sorensor">Medium</a></li>
						<li><a href="about.html">About</a></li>						
					</ul>
				</nav>

				<!-- Main -->
					<div id="main">
						<div class="inner">
							<h1>Predictive Artificial Neural Networks for mRNA Transcriptional Profiles and Sequence Motifs</h1>
							<span class="image header">
								<img src="images/Fig 2 CNN.png" style="width: 1162px;" />
							</span>
							<table>
								<tr>
								  <td>Background</td>
								  <td>
									<p>For more information, see the <a href="https://github.com/Rhys-sg/promoter-sequence-transcription-rates">GitHub repository</a></p>
									<p>Messenger ribonucleic acid (mRNA) synthesis in all organisms is performed by the RNA polymerases, which transcribe corresponding nucleotide base pairs in a gene. The rate of mRNA synthesis is an essential parameter in gene expression. Researchers, such as <a href="https://www.nature.com/articles/s41467-020-19446-w">Van Brempt et al.</a> and <a href="https://www.nature.com/articles/s41467-022-32829-5">LaFleur et al.’s</a> have developed machine learning tools to predict the transcription initiation rates (TX) in E. coli based on RNAP/σ70-promoter interactions. However, there are currently no prediction tools (i) designed for sequential data, (ii) that can account for the complexity of the interplay between regions and employ a modeling method that does not assume independence between sections, and (iii) afford backward-compatible prediction of novel σ70 promoter sequences motifs given a desired TX. This project extends the current research by developing forward- and backward-compatible prediction tools to more accurately predict TX across novel σ70 promoter sequence motifs and novel sequence motifs from TX. The tool uses an artificial neural network (ANN) with a bidirectional information feed between nodes and layers, a state memory, and a multilayer cell structure. The model was optimized using hyperparameter tuning and, compared to the eminent 346-parameter multiple linear regression (MLR), has 81,845 total parameters. The LSTM is “black-boxed” and not as effective for concluding meaningful relationships using parameter analysis. However, it has a significantly lower MSE (0.1400 compared to 0.2515, a decrease of 44%). This model is also the first backward-compatible prediction model, predicting novel sequence motifs given TX, and affording synthetic engineering of non-canonical sequence motifs with desired TX.</p>
								</tr>
								<tr>
									<td>Design</td>
									<td>
										<p>This research has implemented the optimized LSTM model as the back-end for two prediction tools. The first tool is similar to Van Brempt et al. and LaFleur et al.’s and can be used by molecular biologists to more accurately predict genetic context effects, design σ70 promoters with desired transcription rates, and identify undesired promoters inside engineered genetic systems. It works as a wrapper for the model—given a novel sequence and the model file, the tool uses the model and returns a predicted TX given the sequence.</p>
										<p>The second tool is the first backward-compatible prediction model, returning novel sequence motifs with predicted TX closest to a desired value. This tool is more applicable in biophysics and genetic engineering as it can simulate TX of synthetic and non-canonical sequence motifs before engineering them. The tool requires two arguments (the model and desired TX) and has nine optional arguments: the desired TX tolerance, maximum returned results, maximum iterations, and all specified sequence motifs required in the output. The tool calculates every permutation of sequence motifs, predicts a TX value for each permutation, and then returns the sequences with the closest TX. This approach has a runtime of O(M * N!), where N is the sequence length, and M is the number of LSTM parameters. Future research should seek to improve this approach by hashing simulated combinations, applying an inverse transformation, or creating a specialized model. However, all of these approaches still have their respective limitations.</p>
									</td>
								</tr>
							</table>
						</div>
					</div>

				<!-- Footer -->
					<footer id="footer">
						<div class="inner">
							<ul class="copyright">
								<li>Design by <a href="http://html5up.net">HTML5 UP</a></li>
							</ul>
						</div>
					</footer>

			</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>